File: src/main/kotlin/de/keller/physioai/patienten/Patient.kt
package de.keller.physioai.patienten

import java.time.LocalDate

interface Patient {
    val id: PatientId
    val titel: String?
    val vorname: String
    val nachname: String
    val strasse: String?
    val hausnummer: String?
    val plz: String?
    val stadt: String?
    val telMobil: String?
    val telFestnetz: String?
    val email: String?
    val geburtstag: LocalDate?
}

File: frontend/lib/patienten/domain/patient.dart
import 'package:freezed_annotation/freezed_annotation.dart';

part 'generated/patient.freezed.dart';
part 'generated/patient.g.dart';

@freezed
abstract class Patient with _$Patient {
  const factory Patient({
    required String id,
    required String vorname,
    required String nachname,
    required DateTime geburtstag,
    String? titel,
    String? strasse,
    String? hausnummer,
    String? plz,
    String? stadt,
    String? telMobil,
    String? telFestnetz,
    String? email,
  }) = _Patient;

  const Patient._();

  factory Patient.fromJson(Map<String, dynamic> json) => _$PatientFromJson(json);

  String get fullName => [
    if (titel?.isNotEmpty ?? false) titel,
    vorname,
    nachname,
  ].join(' ');

  String get address => [
    [
      if (strasse?.isNotEmpty ?? false) strasse,
      if (hausnummer?.isNotEmpty ?? false) hausnummer,
    ].join(' '),
    [
      if (plz?.isNotEmpty ?? false) plz,
      if (stadt?.isNotEmpty ?? false) stadt,
    ].join(' '),
  ].join('\n');
}
